be/src/olap/rowset/segment_v2/column_reader.cpp

```diff
@@ -338,6 +339,27 @@ bool ColumnReader::match_condition(const AndBlockColumnPredicate* col_predicates
                                      col_predicates);
 }
 
+bool ColumnReader::prune_predicates_by_zone_map(std::vector<ColumnPredicate*>& predicates,
```

Comment:
add a config to disable this
and the prune should only be used in query, not in compaction.